logs/logs.go

```diff
@@ -106,7 +108,15 @@ func (l *LogAgent) Run(ctx context.Context) {
 						log.Printf("E! [logagent] Failed to find destination %v for log source %v/%v(%v) ", dname, src.Group(), src.Stream(), src.Description())
 						continue
 					}
-					dest := backend.CreateDest(src.Group(), src.Stream(), src.Retention())
+					retention := src.Retention()
```

Comment:
Minor thing - There are a bunch of calls to `src.Group()`, `src.Stream()` and `src.Retention()`, and we could limit that by defining the group and stream up here like retention. I doubt that the difference would be noticeable at all between one or the other, but it's also not a difficult thing to swap out.